src/discord-cluster-manager/ui/gpu_selection.py

```diff
@@ -0,0 +1,28 @@
+from discord import Interaction, SelectOption, ui
+
+
+class GPUSelectionView(ui.View):
+    def __init__(self, available_gpus: list[str]):
+        super().__init__()
+
+        # Add the Select Menu with the list of GPU options
+        select = ui.Select(
+            placeholder="Select GPUs for this leaderboard...",
+            options=[SelectOption(label=gpu, value=gpu) for gpu in available_gpus],
+            min_values=1,  # Minimum number of selections
+            max_values=len(available_gpus),  # Maximum number of selections
+        )
+        select.callback = self.select_callback
+        self.add_item(select)
+
+    async def select_callback(self, interaction: Interaction):
+        # Retrieve the selected options
+        select = interaction.data["values"]
+        self.selected_gpus = select
+        # Acknowledge the interaction
+        await interaction.response.defer(ephemeral=True)
+        self.stop()
+
+
+def create_gpu_selection_view(available_gpus: list[str]) -> GPUSelectionView:
```

Comment:
not a fan of shallow functions like this, they justt add indirection and make things harder to understand